src/boost_histogram/_internal/hist.py

```diff
@@ -85,6 +85,22 @@ def _fill_cast(
     return value
 
 
+def meanStorageSampleCheck(sample):
+    assert type(sample) != type(
+        None
+    ), "Sample key-argument (sample=) needs to be provided."
+    assert isinstance(
+        sample, (collections.abc.Sequence, np.ndarray)
+    ) and not isinstance(
+        sample, (str)
+    ), f"Sample key-argument needs to be a sequence, {sample.__class__.__name__} given."
+    assert (
+        np.array(sample).ndim
+    ) == 1, (
+        f"Sample key-argument needs to be 1 dimensional, {np.array(sample).ndim} given."
+    )
```

Comment:
I think assert is probably correct here, but this would be the alternative:

```suggestion
    msg = f"Sample key-argument needs to be 1 dimensional, {np.array(sample).ndim} given."
    if np.array(sample).ndim != 1:
        raise AssertionError(msg)
```